### ISSUE TITLE
[files_antivirus] Upload of Eicar throws Sabre\Exception\UnsupportedMediaType

### ISSUE THREAD

**Issue by michaelstingl** [owncloud/core#30751]
### Steps to reproduce
1. run ownCloud 10.0.7 with `files_antivirus`
2. Upload Eicar

### Expected behaviour
I wouldn't expect the nasty PHP exception  in my logs

### Actual behaviour
```
{"reqId":"MdWK3QiFOS31h9ly9agk","level":4,"time":"2018-03-12T20:11:00+00:00","remoteAddr":"172.18.0.1","user":"admin","app":"webdav","method":"PUT","url":"\/remote.php\/webdav\/eicar-av-test.txt","message":"Exception: {\"Message\":\"HTTP/1.1 415 Virus Eicar-Test-Signature is detected in the file. Upload cannot be completed.\",\"Exception\":\"OCA\DAV\Connector\Sabre\Exception\UnsupportedMediaType\",\"Code\":0,\"Trace\":\"
#0 /var/www/owncloud/apps/dav/lib/Connector/Sabre/File.php(188): OCA\DAV\Connector\Sabre\File->convertToSabreException(Object(OCP\Files\InvalidContentException))\
#1 /var/www/owncloud/lib/public/Events/EventEmitterTrait.php(50): OCA\DAV\Connector\Sabre\File->OCA\DAV\Connector\Sabre\{closure}()\
#2 /var/www/owncloud/apps/dav/lib/Connector/Sabre/File.php(269): OCA\DAV\Connector\Sabre\File->emittingCall(Object(Closure), Array, 'file', 'create')\
#3 /var/www/owncloud/apps/dav/lib/Connector/Sabre/Directory.php(171): OCA\DAV\Connector\Sabre\File->put(Resource id #79)\
#4 /var/www/owncloud/lib/composer/sabre/dav/lib/DAV/Server.php(1095): OCA\DAV\Connector\Sabre\Directory->createFile('eicar-av-test.t...', Resource id #79)\
#5 /var/www/owncloud/lib/composer/sabre/dav/lib/DAV/CorePlugin.php(525): Sabre\DAV\Server->createFile('eicar-av-test.t...', Resource id #79, NULL)\
#6 [internal function]: Sabre\DAV\CorePlugin->httpPut(Object(Sabre\HTTP\Request), Object(Sabre\HTTP\Response))\
#7 /var/www/owncloud/lib/composer/sabre/event/lib/EventEmitterTrait.php(105): call_user_func_array(Array, Array)\
#8 /var/www/owncloud/lib/composer/sabre/dav/lib/DAV/Server.php(479): Sabre\Event\EventEmitter->emit('method:PUT', Array)\
#9 /var/www/owncloud/lib/composer/sabre/dav/lib/DAV/Server.php(254): Sabre\DAV\Server->invokeMethod(Object(Sabre\HTTP\Request), Object(Sabre\HTTP\Response))\
#10 /var/www/owncloud/apps/dav/appinfo/v1/webdav.php(63): Sabre\DAV\Server->exec()\
#11 /var/www/owncloud/remote.php(165): require_once('/var/www/ownclo...')\
#12 {main}\",\"File\":\"/var/www/owncloud/apps/dav/lib/Connector/Sabre/File.php\",\"Line\":632,\"User\":\"admin\"}"}
```

### Server configuration
**Operating system**:
Ubuntu 16.04

**Web server:**
Apache/2.4.18

**Database:**
mysql  Ver 14.14

**PHP version:**
PHP 7.0.25

**ownCloud version:** (see ownCloud admin page)
10.0.7

**Updated from an older ownCloud or fresh install:**
fresh

**Where did you install ownCloud from:**
```
OWNCLOUD_APPS_INSTALL=files_antivirus \
docker-compose \
-f owncloud-base.yml \
-f owncloud-official.yml \
-f owncloud-exported-ports.yml \
-f cache/redis.yml \
-f database/mariadb.yml \
-f services/clamav.yml \
-f configs/enable_files_antivirus_daemon.yml \
up
```

**Signing status (ownCloud 9.0 and above):**

```
No errors have been found.
```

**The content of config/config.php:**

```
{
    "system": {
        "datadirectory": "\/mnt\/data\/files",
        "updatechecker": "false",
        "upgrade.disable-web": true,
        "apps_paths": [
            {
                "path": "\/var\/www\/owncloud\/apps",
                "url": "\/apps",
                "writable": false
            },
            {
                "path": "\/var\/www\/owncloud\/custom",
                "url": "\/custom",
                "writable": true
            }
        ],
        "dbtype": "mysql",
        "dbhost": "db:3306",
        "dbname": "owncloud",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "dbtableprefix": "oc_",
        "trusted_domains": [
            "localhost"
        ],
        "license-key": "***REMOVED SENSITIVE VALUE***",
        "marketplace.key": "266f277976ec9ba02e0ad93510b2795c27ea9d0f6a01579fd9e6d50ed86128ff",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "overwrite.cli.url": "http:\/\/localhost\/",
        "version": "10.0.7.2",
        "logtimezone": "UTC",
        "installed": true,
        "instanceid": "oc1ofjv9fedc",
        "memcache.distributed": "\\OC\\Memcache\\Redis",
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "redis",
            "port": 6379
        },
        "filelocking.enabled": "true",
        "memcache.local": "\\OC\\Memcache\\APCu",
        "loglevel": 0,
        "default_language": "en",
        "htaccess.RewriteBase": "\/"
    }
}
```

**List of activated apps:**

```
Enabled:
  - comments: 0.3.0
  - configreport: 0.1.1
  - dav: 0.3.2
  - enterprise_key: 0.1.4
  - federatedfilesharing: 0.3.1
  - federation: 0.1.0
  - files: 1.5.1
  - files_antivirus: 0.12.0
  - files_external: 0.7.1
  - files_sharing: 0.10.1
  - files_trashbin: 0.9.1
  - files_versions: 1.3.0
  - files_videoplayer: 0.9.8
  - firstrunwizard: 1.1
  - market: 0.2.3
  - notifications: 0.3.2
  - provisioning_api: 0.5.0
  - systemtags: 0.3.0
  - templateeditor: 0.2
  - updatenotification: 0.2.1
Disabled:
  - encryption
  - external
  - theme-example
  - user_external
  - user_ldap
```

**Are you using external storage, if yes which one:** local/smb/sftp/...
no

**Are you using encryption:** yes/no
no

**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...
no

### Client configuration
**Browser:**
Google Chrome Version 64.0.3282.186

**Operating system:**
macOS High Sierra 10.13.3

#### Upload To Web UI (/apps/files/):

```
{"reqId":"MdWK3QiFOS31h9ly9agk","level":0,"time":"2018-03-12T20:11:00+00:00","remoteAddr":"172.18.0.1","user":"admin","app":"files_antivirus","method":"PUT","url":"\/remote.php\/webdav\/eicar-av-test.txt","message":"File size is 68. av_max_file_size is -1. Scanning is scheduled."}
{"reqId":"MdWK3QiFOS31h9ly9agk","level":0,"time":"2018-03-12T20:11:00+00:00","remoteAddr":"172.18.0.1","user":"admin","app":"files_antivirus","method":"PUT","url":"\/remote.php\/webdav\/eicar-av-test.txt","message":"Response :: stream: Eicar-Test-Signature FOUND\n"}
{"reqId":"MdWK3QiFOS31h9ly9agk","level":2,"time":"2018-03-12T20:11:00+00:00","remoteAddr":"172.18.0.1","user":"admin","app":"files_antivirus","method":"PUT","url":"\/remote.php\/webdav\/eicar-av-test.txt","message":"Infected file deleted. Eicar-Test-Signature Account: admin Path: files\/eicar-av-test.txt.ocTransferId60753379.part"}
{"reqId":"MdWK3QiFOS31h9ly9agk","level":4,"time":"2018-03-12T20:11:00+00:00","remoteAddr":"172.18.0.1","user":"admin","app":"webdav","method":"PUT","url":"\/remote.php\/webdav\/eicar-av-test.txt","message":"Exception: {\"Message\":\"HTTP/1.1 415 Virus Eicar-Test-Signature is detected in the file. Upload cannot be completed.\",\"Exception\":\"OCA\DAV\Connector\Sabre\Exception\UnsupportedMediaType\",\"Code\":0,\"Trace\":\"
#0 /var/www/owncloud/apps/dav/lib/Connector/Sabre/File.php(188): OCA\DAV\Connector\Sabre\File->convertToSabreException(Object(OCP\Files\InvalidContentException))\
#1 /var/www/owncloud/lib/public/Events/EventEmitterTrait.php(50): OCA\DAV\Connector\Sabre\File->OCA\DAV\Connector\Sabre\{closure}()\
#2 /var/www/owncloud/apps/dav/lib/Connector/Sabre/File.php(269): OCA\DAV\Connector\Sabre\File->emittingCall(Object(Closure), Array, 'file', 'create')\
#3 /var/www/owncloud/apps/dav/lib/Connector/Sabre/Directory.php(171): OCA\DAV\Connector\Sabre\File->put(Resource id #79)\
#4 /var/www/owncloud/lib/composer/sabre/dav/lib/DAV/Server.php(1095): OCA\DAV\Connector\Sabre\Directory->createFile('eicar-av-test.t...', Resource id #79)\
#5 /var/www/owncloud/lib/composer/sabre/dav/lib/DAV/CorePlugin.php(525): Sabre\DAV\Server->createFile('eicar-av-test.t...', Resource id #79, NULL)\
#6 [internal function]: Sabre\DAV\CorePlugin->httpPut(Object(Sabre\HTTP\Request), Object(Sabre\HTTP\Response))\
#7 /var/www/owncloud/lib/composer/sabre/event/lib/EventEmitterTrait.php(105): call_user_func_array(Array, Array)\
#8 /var/www/owncloud/lib/composer/sabre/dav/lib/DAV/Server.php(479): Sabre\Event\EventEmitter->emit('method:PUT', Array)\
#9 /var/www/owncloud/lib/composer/sabre/dav/lib/DAV/Server.php(254): Sabre\DAV\Server->invokeMethod(Object(Sabre\HTTP\Request), Object(Sabre\HTTP\Response))\
#10 /var/www/owncloud/apps/dav/appinfo/v1/webdav.php(63): Sabre\DAV\Server->exec()\
#11 /var/www/owncloud/remote.php(165): require_once('/var/www/ownclo...')\
#12 {main}\",\"File\":\"/var/www/owncloud/apps/dav/lib/Connector/Sabre/File.php\",\"Line\":632,\"User\":\"admin\"}"}
```

#### Upload To Share Link With Write Permissions (/s/):

```
{"reqId":"kMQNj8xmCs01IXwxYuAI","level":0,"time":"2018-03-12T20:14:11+00:00","remoteAddr":"172.18.0.1","user":"--","app":"files_antivirus","method":"PUT","url":"\/public.php\/webdav\/eicar-av-test.txt","message":"File size is 68. av_max_file_size is -1. Scanning is scheduled."}
{"reqId":"kMQNj8xmCs01IXwxYuAI","level":0,"time":"2018-03-12T20:14:11+00:00","remoteAddr":"172.18.0.1","user":"--","app":"files_antivirus","method":"PUT","url":"\/public.php\/webdav\/eicar-av-test.txt","message":"Response :: stream: Eicar-Test-Signature FOUND\n"}
{"reqId":"kMQNj8xmCs01IXwxYuAI","level":2,"time":"2018-03-12T20:14:11+00:00","remoteAddr":"172.18.0.1","user":"--","app":"files_antivirus","method":"PUT","url":"\/public.php\/webdav\/eicar-av-test.txt","message":"Infected file deleted. Eicar-Test-Signature Account: admin Path: files\/fhxdgchvbn\/eicar-av-test.txt.ocTransferId2058015150.part"}
{"reqId":"kMQNj8xmCs01IXwxYuAI","level":4,"time":"2018-03-12T20:14:11+00:00","remoteAddr":"172.18.0.1","user":"--","app":"webdav","method":"PUT","url":"\/public.php\/webdav\/eicar-av-test.txt","message":"Exception: {\"Message\":\"HTTP/1.1 415 Virus Eicar-Test-Signature is detected in the file. Upload cannot be completed.\",\"Exception\":\"OCA\DAV\Connector\Sabre\Exception\UnsupportedMediaType\",\"Code\":0,\"Trace\":\"
#0 /var/www/owncloud/apps/dav/lib/Connector/Sabre/File.php(188): OCA\DAV\Connector\Sabre\File->convertToSabreException(Object(OCP\Files\InvalidContentException))\
#1 /var/www/owncloud/lib/public/Events/EventEmitterTrait.php(50): OCA\DAV\Connector\Sabre\File->OCA\DAV\Connector\Sabre\{closure}()\
#2 /var/www/owncloud/apps/dav/lib/Connector/Sabre/File.php(269): OCA\DAV\Connector\Sabre\File->emittingCall(Object(Closure), Array, 'file', 'create')\
#3 /var/www/owncloud/apps/dav/lib/Connector/Sabre/Directory.php(171): OCA\DAV\Connector\Sabre\File->put(Resource id #79)\
#4 /var/www/owncloud/lib/composer/sabre/dav/lib/DAV/Server.php(1095): OCA\DAV\Connector\Sabre\Directory->createFile('eicar-av-test.t...', Resource id #79)\
#5 /var/www/owncloud/lib/composer/sabre/dav/lib/DAV/CorePlugin.php(525): Sabre\DAV\Server->createFile('eicar-av-test.t...', Resource id #79, NULL)\
#6 [internal function]: Sabre\DAV\CorePlugin->httpPut(Object(Sabre\HTTP\Request), Object(Sabre\HTTP\Response))\
#7 /var/www/owncloud/lib/composer/sabre/event/lib/EventEmitterTrait.php(105): call_user_func_array(Array, Array)\
#8 /var/www/owncloud/lib/composer/sabre/dav/lib/DAV/Server.php(479): Sabre\Event\EventEmitter->emit('method:PUT', Array)\
#9 /var/www/owncloud/lib/composer/sabre/dav/lib/DAV/Server.php(254): Sabre\DAV\Server->invokeMethod(Object(Sabre\HTTP\Request), Object(Sabre\HTTP\Response))\
#10 /var/www/owncloud/apps/dav/appinfo/v1/publicwebdav.php(95): Sabre\DAV\Server->exec()\
#11 /var/www/owncloud/public.php(76): require_once('/var/www/ownclo...')\
#12 {main}\",\"File\":\"/var/www/owncloud/apps/dav/lib/Connector/Sabre/File.php\",\"Line\":632,\"User\":false}"}
{"reqId":"MojLJHfoHXW0OzqEbePV","level":0,"time":"2018-03-12T20:15:01+00:00","remoteAddr":"","user":"--","app":"cron","method":"--","url":"--","message":"Started background job of class : OC\\Authentication\\Token\\DefaultTokenCleanupJob with arguments : "}
```

#### Upload Using ownCloud Desktop Sync Client 2.4.1 (new "dav" endpoint):

```
{"reqId":"d8079abb-9e88-43e3-aea2-377aab799271","level":0,"time":"2018-03-12T20:17:03+00:00","remoteAddr":"172.18.0.1","user":"admin","app":"files_antivirus","method":"PUT","url":"\/remote.php\/dav\/files\/admin\/eicar-av-test.txt","message":"File size is 68. av_max_file_size is -1. Scanning is scheduled."}
{"reqId":"d8079abb-9e88-43e3-aea2-377aab799271","level":0,"time":"2018-03-12T20:17:03+00:00","remoteAddr":"172.18.0.1","user":"admin","app":"files_antivirus","method":"PUT","url":"\/remote.php\/dav\/files\/admin\/eicar-av-test.txt","message":"Response :: stream: Eicar-Test-Signature FOUND\n"}
{"reqId":"d8079abb-9e88-43e3-aea2-377aab799271","level":2,"time":"2018-03-12T20:17:03+00:00","remoteAddr":"172.18.0.1","user":"admin","app":"files_antivirus","method":"PUT","url":"\/remote.php\/dav\/files\/admin\/eicar-av-test.txt","message":"Infected file deleted. Eicar-Test-Signature Account: admin Path: files\/eicar-av-test.txt.ocTransferId462077176.part"}
{"reqId":"d8079abb-9e88-43e3-aea2-377aab799271","level":4,"time":"2018-03-12T20:17:03+00:00","remoteAddr":"172.18.0.1","user":"admin","app":"webdav","method":"PUT","url":"\/remote.php\/dav\/files\/admin\/eicar-av-test.txt","message":"Exception: {\"Message\":\"HTTP/1.1 415 Virus Eicar-Test-Signature is detected in the file. Upload cannot be completed.\",\"Exception\":\"OCA\DAV\Connector\Sabre\Exception\UnsupportedMediaType\",\"Code\":0,\"Trace\":\"
#0 /var/www/owncloud/apps/dav/lib/Connector/Sabre/File.php(188): OCA\DAV\Connector\Sabre\File->convertToSabreException(Object(OCP\Files\InvalidContentException))\
#1 /var/www/owncloud/lib/public/Events/EventEmitterTrait.php(50): OCA\DAV\Connector\Sabre\File->OCA\DAV\Connector\Sabre\{closure}()\
#2 /var/www/owncloud/apps/dav/lib/Connector/Sabre/File.php(269): OCA\DAV\Connector\Sabre\File->emittingCall(Object(Closure), Array, 'file', 'create')\
#3 /var/www/owncloud/apps/dav/lib/Connector/Sabre/Directory.php(171): OCA\DAV\Connector\Sabre\File->put(Resource id #79)\
#4 /var/www/owncloud/lib/composer/sabre/dav/lib/DAV/Server.php(1095): OCA\DAV\Connector\Sabre\Directory->createFile('eicar-av-test.t...', Resource id #79)\
#5 /var/www/owncloud/lib/composer/sabre/dav/lib/DAV/CorePlugin.php(525): Sabre\DAV\Server->createFile('files/admin/eic...', Resource id #79, NULL)\
#6 [internal function]: Sabre\DAV\CorePlugin->httpPut(Object(Sabre\HTTP\Request), Object(Sabre\HTTP\Response))\
#7 /var/www/owncloud/lib/composer/sabre/event/lib/EventEmitterTrait.php(105): call_user_func_array(Array, Array)\
#8 /var/www/owncloud/lib/composer/sabre/dav/lib/DAV/Server.php(479): Sabre\Event\EventEmitter->emit('method:PUT', Array)\
#9 /var/www/owncloud/lib/composer/sabre/dav/lib/DAV/Server.php(254): Sabre\DAV\Server->invokeMethod(Object(Sabre\HTTP\Request), Object(Sabre\HTTP\Response))\
#10 /var/www/owncloud/apps/dav/lib/Server.php(263): Sabre\DAV\Server->exec()\
#11 /var/www/owncloud/apps/dav/appinfo/v2/remote.php(31): OCA\DAV\Server->exec()\
#12 /var/www/owncloud/remote.php(165): require_once('/var/www/ownclo...')\
#13 {main}\",\"File\":\"/var/www/owncloud/apps/dav/lib/Connector/Sabre/File.php\",\"Line\":632,\"User\":\"admin\"}"}
```

#### Upload Using WebDAV Client (old "webdav" endpoint):

```
{"reqId":"tTfuNd0IiPrxnWBZO94m","level":0,"time":"2018-03-12T20:15:26+00:00","remoteAddr":"172.18.0.1","user":"admin","app":"files_antivirus","method":"PUT","url":"\/remote.php\/dav\/files\/admin\/eicar-av-test.txt","message":"File size is 68. av_max_file_size is -1. Scanning is scheduled."}
{"reqId":"tTfuNd0IiPrxnWBZO94m","level":0,"time":"2018-03-12T20:15:26+00:00","remoteAddr":"172.18.0.1","user":"admin","app":"files_antivirus","method":"PUT","url":"\/remote.php\/dav\/files\/admin\/eicar-av-test.txt","message":"Response :: stream: Eicar-Test-Signature FOUND\n"}
{"reqId":"tTfuNd0IiPrxnWBZO94m","level":2,"time":"2018-03-12T20:15:26+00:00","remoteAddr":"172.18.0.1","user":"admin","app":"files_antivirus","method":"PUT","url":"\/remote.php\/dav\/files\/admin\/eicar-av-test.txt","message":"Infected file deleted. Eicar-Test-Signature Account: admin Path: files\/eicar-av-test.txt.ocTransferId945058288.part"}
{"reqId":"tTfuNd0IiPrxnWBZO94m","level":4,"time":"2018-03-12T20:15:26+00:00","remoteAddr":"172.18.0.1","user":"admin","app":"webdav","method":"PUT","url":"\/remote.php\/dav\/files\/admin\/eicar-av-test.txt","message":"Exception: {\"Message\":\"HTTP/1.1 415 Virus Eicar-Test-Signature is detected in the file. Upload cannot be completed.\",\"Exception\":\"OCA\DAV\Connector\Sabre\Exception\UnsupportedMediaType\",\"Code\":0,\"Trace\":\"
#0 /var/www/owncloud/apps/dav/lib/Connector/Sabre/File.php(188): OCA\DAV\Connector\Sabre\File->convertToSabreException(Object(OCP\Files\InvalidContentException))\
#1 /var/www/owncloud/lib/public/Events/EventEmitterTrait.php(50): OCA\DAV\Connector\Sabre\File->OCA\DAV\Connector\Sabre\{closure}()\
#2 /var/www/owncloud/apps/dav/lib/Connector/Sabre/File.php(269): OCA\DAV\Connector\Sabre\File->emittingCall(Object(Closure), Array, 'file', 'create')\
#3 /var/www/owncloud/apps/dav/lib/Connector/Sabre/Directory.php(171): OCA\DAV\Connector\Sabre\File->put(Resource id #79)\
#4 /var/www/owncloud/lib/composer/sabre/dav/lib/DAV/Server.php(1095): OCA\DAV\Connector\Sabre\Directory->createFile('eicar-av-test.t...', Resource id #79)\
#5 /var/www/owncloud/lib/composer/sabre/dav/lib/DAV/CorePlugin.php(525): Sabre\DAV\Server->createFile('files/admin/eic...', Resource id #79, NULL)\
#6 [internal function]: Sabre\DAV\CorePlugin->httpPut(Object(Sabre\HTTP\Request), Object(Sabre\HTTP\Response))\
#7 /var/www/owncloud/lib/composer/sabre/event/lib/EventEmitterTrait.php(105): call_user_func_array(Array, Array)\
#8 /var/www/owncloud/lib/composer/sabre/dav/lib/DAV/Server.php(479): Sabre\Event\EventEmitter->emit('method:PUT', Array)\
#9 /var/www/owncloud/lib/composer/sabre/dav/lib/DAV/Server.php(254): Sabre\DAV\Server->invokeMethod(Object(Sabre\HTTP\Request), Object(Sabre\HTTP\Response))\
#10 /var/www/owncloud/apps/dav/lib/Server.php(263): Sabre\DAV\Server->exec()\
#11 /var/www/owncloud/apps/dav/appinfo/v2/remote.php(31): OCA\DAV\Server->exec()\
#12 /var/www/owncloud/remote.php(165): require_once('/var/www/ownclo...')\
#13 {main}\",\"File\":\"/var/www/owncloud/apps/dav/lib/Connector/Sabre/File.php\",\"Line\":632,\"User\":\"admin\"}"}
```

@felixboehm suspects https://github.com/owncloud/core/commit/74f8be86d4bbba9b3e6137089aa2fea1fde4dfe7#diff-be504dc886734275f06114e4306fa215 to cause this.

00009091

**Comment by ownclouders** [comment]
[GitMate.io](https://gitmate.io) thinks the contributor most likely able to help you is @PVince81.

Possibly related issues are https://github.com/owncloud/core/issues/18166 (Upload deduplication), https://github.com/owncloud/core/issues/12658 (Upload-only feature), https://github.com/owncloud/core/issues/18111 ([feat] Upload from URL), https://github.com/owncloud/core/issues/14352 (Permission to upload), and https://github.com/owncloud/core/issues/24678 (NotFoundException thrown randomly).

**Comment by PVince81** [comment]
@sharidas can you debug ?

**Comment by PVince81** [comment]
@michaelstingl is the message still propagated to the clients or does it just stop at the logged exception ?

**Comment by michaelstingl** [comment]
I think it looks good in the clients. Only the logs scared me.

**Comment by sharidas** [comment]
> @sharidas can you debug ?

Indeed I will.

**Comment by VicDeo** [comment]
FYI https://github.com/owncloud/files_antivirus/issues/177

**Comment by sharidas** [comment]
The purpose of this exception is to stop the upload. That's the intention I find from the codeflow. Need to figure out how to regulate the exception.

**Comment by PVince81** [comment]
@sharidas my worry is mostly that the commit in question which brings in new hook format has changed the behavior. It shouldn't change behavior at all.

**Comment by sharidas** [comment]
When I try to checkout commit : https://github.com/owncloud/core/commit/b907afe0e91e469b479c690dd11867b063f1bf08 , which is the commit before ( in the git log it comes under the suspected commit in this issue i.e, https://github.com/owncloud/core/commit/74f8be86d4bbba9b3e6137089aa2fea1fde4dfe7#diff-be504dc886734275f06114e4306fa215 ).  The oc version becomes 10.0.4 and I can still find exception. 
![gifrecord_2018-03-13_143023](https://user-images.githubusercontent.com/3600427/37331971-76d0fd8e-26cb-11e8-9d47-729cdfb2fca4.gif)

**Comment by sharidas** [comment]
It works in 9.1.8. Will try to bisect and figure out which pr caused this regression.

**Comment by sharidas** [comment]
I find that https://github.com/owncloud/core/commit/540987e3209d3a1354117ecc766628db85a46d12 had caused the change to log exception. This finding is based on git bisect.

**Comment by PVince81** [comment]
@michaelstingl when was it last working correctly ?

**Comment by PVince81** [comment]
Does uploading with Webdav PUT on the CLI also trigger the exception ?

The change that 540987e made was mostly to make the web UI use Webdav PUT for uploads.
Maybe this bug already existed before.

**Comment by sharidas** [comment]
When I try from the CLI:
```
➜  owncloud3 git:(master) ✗ curl -u admin:admin -T /tmp/eicar.com.txt "http://localhost/testing3/remote.php/dav/files/admin/eicar.com.txt"
<?xml version="1.0" encoding="utf-8"?>
<d:error xmlns:d="DAV:" xmlns:s="http://sabredav.org/ns">
  <s:exception>Sabre\DAV\Exception</s:exception>
  <s:message>Virus Eicar-Test-Signature is detected in the file. Upload cannot be completed.</s:message>
</d:error>
➜  owncloud3 git:(master) ✗
```
The exception is triggered:
```
{"reqId":"P8wINDeF5l0i9R8Yurq4","level":4,"time":"2018-03-13T14:48:14+00:00","remoteAddr":"127.0.0.1","user":"admin","app":"webdav","method":"PUT","url":"\/testing3\/remote.php\/dav\/files\/admin\/eicar.com.txt","message":"Exception: {\"Message\":\"HTTP\\\/1.1 500 Virus Eicar-Test-Signature is detected in the file. Upload cannot be completed.\",\"Exception\":\"Sabre\\\\DAV\\\\Exception\",\"Code\":0,\"Trace\":\"#0 \\\/home\\\/sujith\\\/test\\\/owncloud3\\\/apps\\\/dav\\\/lib\\\/Connector\\\/Sabre\\\/File.php(189): OCA\\\\DAV\\\\Connector\\\\Sabre\\\\File->convertToSabreException(Object(OCP\\\\Files\\\\ExcludeForbiddenException))\\n#1 \\\/home\\\/sujith\\\/test\\\/owncloud3\\\/lib\\\/public\\\/Events\\\/EventEmitterTrait.php(50): OCA\\\\DAV\\\\Connector\\\\Sabre\\\\File->OCA\\\\DAV\\\\Connector\\\\Sabre\\\\{closure}()\\n#2 \\\/home\\\/sujith\\\/test\\\/owncloud3\\\/apps\\\/dav\\\/lib\\\/Connector\\\/Sabre\\\/File.php(258): OCA\\\\DAV\\\\Connector\\\\Sabre\\\\File->emittingCall(Object(Closure), Array, 'file', 'create')\\n#3 \\\/home\\\/sujith\\\/test\\\/owncloud3\\\/apps\\\/dav\\\/lib\\\/Connector\\\/Sabre\\\/Directory.php(172): OCA\\\\DAV\\\\Connector\\\\Sabre\\\\File->put(Resource id #56)\\n#4 \\\/home\\\/sujith\\\/test\\\/owncloud3\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(1096): OCA\\\\DAV\\\\Connector\\\\Sabre\\\\Directory->createFile('eicar.com.txt', Resource id #56)\\n#5 \\\/home\\\/sujith\\\/test\\\/owncloud3\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/CorePlugin.php(525): Sabre\\\\DAV\\\\Server->createFile('files\\\/admin\\\/eic...', Resource id #56, NULL)\\n#6 [internal function]: Sabre\\\\DAV\\\\CorePlugin->httpPut(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#7 \\\/home\\\/sujith\\\/test\\\/owncloud3\\\/lib\\\/composer\\\/sabre\\\/event\\\/lib\\\/EventEmitterTrait.php(105): call_user_func_array(Array, Array)\\n#8 \\\/home\\\/sujith\\\/test\\\/owncloud3\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(480): Sabre\\\\Event\\\\EventEmitter->emit('method:PUT', Array)\\n#9 \\\/home\\\/sujith\\\/test\\\/owncloud3\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(255): Sabre\\\\DAV\\\\Server->invokeMethod(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#10 \\\/home\\\/sujith\\\/test\\\/owncloud3\\\/apps\\\/dav\\\/lib\\\/Server.php(290): Sabre\\\\DAV\\\\Server->exec()\\n#11 \\\/home\\\/sujith\\\/test\\\/owncloud3\\\/apps\\\/dav\\\/appinfo\\\/v2\\\/remote.php(31): OCA\\\\DAV\\\\Server->exec()\\n#12 \\\/home\\\/sujith\\\/test\\\/owncloud3\\\/remote.php(175): require_once('\\\/home\\\/sujith\\\/te...')\\n#13 {main}\",\"File\":\"\\\/home\\\/sujith\\\/test\\\/owncloud3\\\/apps\\\/dav\\\/lib\\\/Connector\\\/Sabre\\\/File.php\",\"Line\":640,\"User\":\"admin\"}"}
```

**Comment by sharidas** [comment]
One idea which worked is: to have an exception named `ExcludeForbiddenException` and then replace https://github.com/owncloud/files_antivirus/blob/master/lib/AvirWrapper.php#L181 with `ExcludeForbiddenException`. And then if the previous exception is `ExcludeForbiddenException` then return https://github.com/owncloud/core/blob/master/apps/dav/lib/Connector/Sabre/ExceptionLoggerPlugin.php#L89
This way I believe we wont be breaking the code flow, but just skipping the logging.
Let me know if this solution looks fine?

**Comment by sharidas** [comment]
This is what I meant: https://github.com/owncloud/core/pull/30771

**Comment by VicDeo** [comment]
yes, fixed by @sharidas  in 10.0.8
